content/blog/2021-11-10-agent.md

```diff
@@ -0,0 +1,156 @@
+
+> Wait for a second, Bartek. You just mentioned before that pushing metrics directly from the application is not the best idea!
+
+Sure, but the amazing part is that, even with the Remote Write, Prometheus still uses a pull model to gather metrics from applications, which gives us an understanding of those different failure modes. After that, we batch samples and series and export, replicate (push) data to the Remote Write endpoints, limiting the number of monitoring unknowns central point has!
+
+It's important to note that a reliable and efficient remote-writing implementation is a non-trivial problem to solve. The Prometheus community spent around three years to come with a stable and scalable implementation. We reimplemented WAL (Write-Ahead-Log) a few times, added internal queuing, sharding, smart back-offs and more. All of this is hidden from the user who can enjoy well-performing streaming or large amounts of metrics off clusters.
+
+### Hands-on Remote Write Example: Katacoda Tutorial
+
+All of this is not new in Prometheus. Many of us already use Prometheus to scrape all required metrics and remote-write all or some of them to remote locations.
+
+Suppose you would like to try the hands-on experience of remote writing capabilities. In that case, we recommend our [Thanos Katacoda tutorial of remote writing metrics from Prometheus](https://katacoda.com/thanos/courses/thanos/3-receiver), which explains all steps required for Prometheus to forward all metrics to the remote location. It's **free**, just sign up for an account and enjoy the tutorial! 🤗
+
+Note that this example uses Thanos in receive mode as the remote storage. Nowadays, you can use plenty of other projects that are compatible with remote write API.
+
+So if remote writing works fine, why do we add special Agent mode to Prometheus?
+
+## Prometheus Agent Mode
+
+From Prometheus `v2.32.0` (next release), everyone will be able to run Prometheus binary with an experimental `--enable-feature=agent` flag. If you want to try it before the release, feel free to use [Prometheus v2.32.0-beta.0](https://github.com/prometheus/prometheus/releases/tag/v2.32.0-beta.0) or use our `quay.io/prometheus/prometheus:v2.32.0-beta.0` image.
+
+The Agent mode optimizes Prometheus for the remote write use case. It disables querying, alerting and local storage and replaces it with customized TSDB WAL. Everything else stays the same: scraping logic, service discovery and related configuration. It can be used as a drop-in replacement for Prometheus if you want to just forward your data to remote Prometheus and any other Remote Write compliant project. In essence it looks like this:
```

Comment:
```suggestion
The Agent mode optimizes Prometheus for the remote write use case. It disables querying, alerting, and local storage, and replaces it with a customized TSDB WAL. Everything else stays the same: scraping logic, service discovery and related configuration. It can be used as a drop-in replacement for Prometheus if you want to just forward your data to a remote Prometheus server or any other Remote-Write-compliant project. In essence it looks like this:
```